### ISSUE TITLE
Hint Analyzer does not recognize 'gradle' entries

### ISSUE THREAD

**Issue by isaguimiot** [dependency-check/DependencyCheck#1570]
Hi !

I have a very weird issue I discovered through one of our sample apps that uses both maven and gradle : the vulnerability CVE-2018-15758 (which, in our case, is NOT a false positive) is raised by the maven plugin, but not by the gradle plugin. I made a small sample to reproduce the issue (see attachment), and I also tested the dependency-check cli on it : it doesn't find it either...

As you can see in my sample, I use org.springframework.security.oauth:spring-security-oauth2:2.2.0.RELEASE , which is lower than 2.2.3, so it should raise 2018-15758.

I always use the central analyzer for gradle, maven and the cli. I'm using the latest version 3.3.4 on the 3 of them...

Is there a problem in my configuration ? How is it possible, they all use the same core engine, how can we get different issues on the same set of libraries ?

Thanks for your help !

Isabelle

[basic-project-issue-CVE-2018-15758.zip](https://github.com/jeremylong/DependencyCheck/files/2600827/basic-project-issue-CVE-2018-15758.zip)

**Comment by jeremylong** [comment]
Thank you for identifying this bug. The hint analyzer specifically looks for "pom" entries in the evidence. However, in the case of gradle - those same entries are under a 'gradle' entry instead of under the pom entry.  Should be a simple fix.

As to why the CLI would not find this is because the evidence collected is only from the file itself - no information is captured from the build tool itself. We generally recommend that people use the maven or gradle plugins (or sbt plugin).

**Comment by isaguimiot** [comment]
Hi Jeremy ! 

Have you planned to work on this issue in the next weeks ?

Thank you,
Isabelle

**Comment by isaguimiot** [comment]
Hi Jeremy,

I made a few tests with dependency-check 4.0.2, and now I noticed that CVE-2018-15758 is now raised by both maven and gradle. Does it mean that you fixed this issue and forgot to close it ? Or is it just a coincidence ? Maybe the other improvements you made in version 4.x better detect CVEs in gradle, and the hint analyzer has become useless ?

Just let me know !

Thanks :)

**Comment by jeremylong** [comment]
Sorry for the much delayed response - too many open tickets.  Yes, changes were made in the 4.x branch that improve the analysis but also increased the number of false positives.  We have since improved the analysis more in the 5.x branch and we have also reduced the false positives that were introduced in the 4.x branch.  We hope to have 5.0.0 release in the next week - but may get delayed due to travel or someone finding additional severe defects in the 5.0.0-M3 release.
